src/gprs.c

```diff
@@ -536,9 +542,181 @@ static void pri_context_signal_settings(struct pri_context *ctx,
 				context_settings_append_ipv6);
 }
 
-static void pri_parse_proxy(struct pri_context *ctx, const char *proxy)
+static void set_route(const struct context_settings *settings,
+					const char *ipstr, gboolean create)
+{
+	struct rtentry rt;
+	struct sockaddr_in addr;
+	int sk;
+
+	/* TODO Handle IPv6 case */
+
+	DBG("%s for %s", ipstr, create ? "create" : "remove");
+
+	if (settings->interface == NULL)
+		return;
+
+	sk = socket(PF_INET, SOCK_DGRAM, 0);
+	if (sk < 0)
+		return;
+
+	memset(&rt, 0, sizeof(rt));
+	rt.rt_dev = (char *) settings->interface;
+	rt.rt_flags = RTF_HOST;
+
+	if (create)
+		rt.rt_flags |= RTF_UP;
+
+	memset(&addr, 0, sizeof(addr));
+	addr.sin_family = AF_INET;
+	addr.sin_addr.s_addr = inet_addr(ipstr);
+	if (addr.sin_addr.s_addr == INADDR_NONE) {
+		ofono_error("Cannot delete route for invalid IP %s", ipstr);
+		return;
+	}
+	memcpy(&rt.rt_dst, &addr, sizeof(rt.rt_dst));
+
+	memset(&addr, 0, sizeof(addr));
+	addr.sin_family = AF_INET;
+	addr.sin_addr.s_addr = INADDR_ANY;
+	memcpy(&rt.rt_gateway, &addr, sizeof(rt.rt_gateway));
+	memcpy(&rt.rt_genmask, &addr, sizeof(rt.rt_genmask));
+
+	if (ioctl(sk, create ? SIOCADDRT : SIOCDELRT, &rt) < 0)
+		ofono_error("Failed to delete proxy host route: %s (%d)",
```

Comment:
This log message needs an extra string parameter set via: 'create ? "create" : "remove"'.